flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/restore/RocksDBIncrementalRestoreOperation.java

```diff
@@ -461,26 +457,18 @@ private RestoredDBInstance restoreDBInstanceFromStateHandle(
 	 * a local state.
 	 */
 	private void restoreInstanceDirectoryFromPath(Path source, String instanceRocksDBPath) throws IOException {
+		final Path instanceRocksDBDirectory = Paths.get(instanceRocksDBPath);
+		final Path[] files = Files.list(source).toArray(Path[]::new);
```

Comment:
same as above

Reply:
this could also be done as a `map` operation on the `Stream<Path>`, then we wouldn't need the array allocation

Reply:
The problem with `.map(lambda)` is that some methods in there throw checked exceptions.
I think array/iteration is simpler in the end than exception wrapping / unwrapping.
It is also a rather rare operation, so the occasional array allocation should be fine.